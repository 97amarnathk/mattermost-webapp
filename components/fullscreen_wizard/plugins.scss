@import 'utils/variables';
@import 'utils/mixins';
@import './fullscreen_wizard';

.Plugins-body {
    @include absolutely-fullscreen;
}

@include simple-in-and-out("Plugins");
