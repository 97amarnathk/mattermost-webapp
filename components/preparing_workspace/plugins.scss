@import 'utils/mixins';

.Plugins-body {
    display: flex;
}

.Plugins__marketplace {
    margin-top: 24px;
}

// width at which the cards start hitting screen and wrapping to next line.
@media screen and (max-width: 1000px) {
    .Plugins-body .MultiSelectCards {
        display: flex;
        flex-direction: column;
        align-items: center;
        margin-right: 16px;

        .MultiSelectCard {
            width: 100%;
            max-width: 500px;
            flex-grow: 1;
            margin-right: 0;
            margin-bottom: 16px;

            &:last-child {
                margin-bottom: 0;
            }
        }
    }
}

@include simple-in-and-out("Plugins");
