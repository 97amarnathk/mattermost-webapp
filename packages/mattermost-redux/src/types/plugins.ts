// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

// TODO remove
type Plugin = {
    id: string;
    name: string;
    description?: string;
    homepage_url?: string;
    support_url?: string;
    release_notes_url?: string;
    icon_path?: string;
    version: string;
    min_server_version?: string;
    server?: PluginManifestServer;
    backend?: PluginManifestServer;
    webapp?: PluginManifestWebapp;
    settings_schema?: PluginSettingsSchema;
    props?: Record<string, any>;
};

// TODO remove
export type PluginManifest = Plugin;

export type PluginRedux = PluginManifest & {active: boolean};

// TODO remove
export type PluginManifestServer = {
    executables?: {
        'linux-amd64'?: string;
        'darwin-amd64'?: string;
        'windows-amd64'?: string;
    };
    executable: string;
};

// TODO remove
export type PluginManifestWebapp = {
    bundle_path: string;
};

// TODO remove
export type PluginSettingsSchema = {
    header: string;
    footer: string;
    settings: PluginSetting[];
};

// TODO remove
export type PluginSetting = {
    key: string;
    display_name: string;
    type: string;
    help_text: string;
    regenerate_help_text?: string;
    placeholder: string;
    default: any;
    options?: PluginSettingOption[];
};

// TODO remove
export type PluginSettingOption = {
    display_name: string;
    value: string;
};

// TODO remove
export type PluginsResponse = {
    active: PluginManifest[];
    inactive: PluginManifest[];
};

// TODO remove
export type PluginStatus = {
    plugin_id: string;
    cluster_id: string;
    plugin_path: string;
    state: number;
    name: string;
    description: string;
    version: string;
};

type PluginInstance = {
    cluster_id: string;
    version: string;
    state: number;
}

export type PluginStatusRedux = {
    id: string;
    name: string;
    description: string;
    version: string;
    active: boolean;
    state: number;
    instances: PluginInstance[];
}

// TODO remove
export type ClientPluginManifest = {
    id: string;
    min_server_version?: string;
    version: string;
    webapp: {
        bundle_path: string;
    };
}

export type MarketplaceLabel = { // TODO remove this in favour of the definition in types/marketplace after the mattermost-redux migration
    name: string;
    description?: string;
    url?: string;
    color?: string;
}

export enum HostingType { // TODO remove this in favour of the definition in types/marketplace after the mattermost-redux migration
    OnPrem = 'on-prem',
    Cloud = 'cloud',
}

export enum AuthorType { // TODO remove this in favour of the definition in types/marketplace after the mattermost-redux migration
    Mattermost = 'mattermost',
    Partner = 'partner',
    Community = 'community',
}

export enum ReleaseStage { // TODO remove this in favour of the definition in types/marketplace after the mattermost-redux migration
    Production = 'production',
    Beta = 'beta',
    Experimental = 'experimental',
}

export type MarketplacePlugin = { // TODO remove this in favour of the definition in types/marketplace after the mattermost-redux migration
    homepage_url?: string;
    icon_data?: string;
    download_url?: string;
    release_notes_url?: string;
    labels?: MarketplaceLabel[];
    hosting?: HostingType;
    author_type: AuthorType;
    release_stage: ReleaseStage;
    enterprise: boolean;
    manifest: PluginManifest;
    installed_version?: string;
}
