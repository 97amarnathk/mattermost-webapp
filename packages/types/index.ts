// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export * from './src/admin';
export * from './src/apps';
export * from './src/audit';
export * from './src/autocomplete';
export * from './src/bots';
export * from './src/channel_categories';
export * from './src/channels';
export * from './src/cloud';
export * from './src/compliance';
export * from './src/config';
export * from './src/data_retention';
export * from './src/emojis';
export * from './src/errors';
export * from './src/files';
export * from './src/general';
export * from './src/groups';
export * from './src/integration_actions';
export * from './src/integrations';
export * from './src/jobs';
export * from './src/marketplace';
export * from './src/message_attachments';
export * from './src/mfa';
export * from './src/plugins';
export * from './src/posts';
export * from './src/preferences';
export * from './src/product_notices';
export * from './src/reactions';
export * from './src/roles';
export * from './src/saml';
export * from './src/schemes';
export * from './src/sessions';
export * from './src/teams';
export * from './src/terms_of_service';
export * from './src/threads';
export * from './src/users';
