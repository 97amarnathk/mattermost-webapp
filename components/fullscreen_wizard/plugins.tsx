// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {FormattedMessage, useIntl} from 'react-intl';
import {CSSTransition} from 'react-transition-group';

import {Animations, mapAnimationReasonToClass, Form, TransitionProps} from './steps';

import WizardRadioButton from './wizard_radio_button';

import GithubSVG from './github.svg';
import GitlabSVG from './gitlab.svg';
import JiraSVG from './jira.svg';
import ZoomSVG from './zoom.svg';
import TodoSVG from './todo.svg';

import PageLine from './page_line';

import './plugins.scss';

type Props = TransitionProps & {
    options: Form['plugins'];
    setOption: (option: keyof Form['plugins']) => void;
}
const Plugins = (props: Props) => {
    const {formatMessage} = useIntl();
    return (
        <CSSTransition
            in={props.show}
            timeout={Animations.PAGE_SLIDE}
            classNames={mapAnimationReasonToClass('Plugins', props.direction)}
            mountOnEnter={true}
            unmountOnExit={true}
        >
            <div className='Plugins-body'>
                <PageLine height={'100px'}/>
                {props.previous}
                {
                    <ul>
                        <WizardRadioButton
                            onClick={() => props.setOption('github')}
                            icon={<GithubSVG/>}
                            msg={
                                <FormattedMessage
                                    id={'onboarding_wizard.plugins.github'}
                                    defaultMessage='Github'
                                />
                            }
                            checked={props.options.github}
                            tooltip={formatMessage({
                                id: 'onboarding_wizard.plugins.github.tooltip',
                                defaultMessage: 'Subscribe to repositories, stay up to date with reviews, assignments',
                            })}
                        />
                        <WizardRadioButton
                            onClick={() => props.setOption('gitlab')}
                            icon={<GitlabSVG/>}
                            msg={
                                <FormattedMessage
                                    id={'onboarding_wizard.plugins.gitlab'}
                                    defaultMessage='Gitlab'
                                />
                            }
                            checked={props.options.gitlab}
                            tooltip={formatMessage({
                                id: 'onboarding_wizard.plugins.gitlab.tooltip',
                                defaultMessage: 'Gitlab tooltip',
                            })}
                        />
                        <WizardRadioButton
                            onClick={() => props.setOption('jira')}
                            icon={<JiraSVG/>}
                            msg={
                                <FormattedMessage
                                    id={'onboarding_wizard.plugins.jira'}
                                    defaultMessage='Jira'
                                />
                            }
                            checked={props.options.jira}
                            tooltip={formatMessage({
                                id: 'onboarding_wizard.plugins.jira.tooltip',
                                defaultMessage: 'Jira tooltip',
                            })}
                        />
                        <WizardRadioButton
                            onClick={() => props.setOption('zoom')}
                            icon={<ZoomSVG/>}
                            msg={
                                <FormattedMessage
                                    id={'onboarding_wizard.plugins.zoom'}
                                    defaultMessage='Zoom'
                                />
                            }
                            checked={props.options.zoom}
                            tooltip={formatMessage({
                                id: 'onboarding_wizard.plugins.zoom.tooltip',
                                defaultMessage: 'Zoom tooltip',
                            })}
                        />
                        <WizardRadioButton
                            onClick={() => props.setOption('todo')}
                            icon={<TodoSVG/>}
                            msg={
                                <FormattedMessage
                                    id={'onboarding_wizard.plugins.todo'}
                                    defaultMessage='To do'
                                />
                            }
                            checked={props.options.todo}
                            tooltip={formatMessage({
                                id: 'onboarding_wizard.plugins.todo.tooltip',
                                defaultMessage: 'To do tooltip',
                            })}
                        />
                    </ul>
                }
                <button
                    className='btn btn-primary'
                    onClick={props.next}
                >
                    <FormattedMessage
                        id={'onboarding_wizard.next'}
                        defaultMessage='Continue'
                    />
                </button>
                <button
                    className='tertiary-button'
                    onClick={props.skip}
                >
                    <FormattedMessage
                        id={'onboarding_wizard.skip'}
                        defaultMessage='Skip for now'
                    />
                </button>
                <PageLine height={'calc(100vh - 100px - 400px'}/>
            </div>
        </CSSTransition>
    );
};

export default Plugins;
