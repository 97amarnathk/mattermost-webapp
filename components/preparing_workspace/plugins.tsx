// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {FormattedMessage, useIntl} from 'react-intl';
import {CSSTransition} from 'react-transition-group';

import {t} from 'utils/i18n';
import MultiSelectCards from 'components/common/multi_select_cards';

import FormattedMarkdownMessage from 'components/formatted_markdown_message';

import GithubSVG from 'components/common/svg_images_components/github_svg';
import GitlabSVG from 'components/common/svg_images_components/gitlab_svg';
import JiraSVG from 'components/common/svg_images_components/jira_svg';
import ZoomSVG from 'components/common/svg_images_components/zoom_svg';
import TodoSVG from 'components/common/svg_images_components/todo_svg';

import {Animations, mapAnimationReasonToClass, Form, TransitionProps} from './steps';

import PageLine from './page_line';

import './plugins.scss';

type Props = TransitionProps & {
    options: Form['plugins'];
    setOption: (option: keyof Form['plugins']) => void;
    className?: string;
}
const Plugins = (props: Props) => {
    const {formatMessage} = useIntl();
    let className = 'Plugins-body';

    if (props.className) {
        className += ' ' + props.className;
    }
    return (
        <CSSTransition
            in={props.show}
            timeout={Animations.PAGE_SLIDE}
            classNames={mapAnimationReasonToClass('Plugins', props.direction)}
            mountOnEnter={true}
            unmountOnExit={true}
        >
            <div className={className}>
                <PageLine style={{height: '100px'}}/>
                {props.previous}
                {
                    <MultiSelectCards
                        next={props.next}
                        cards={[
                            {
                                onClick: () => props.setOption('github'),
                                icon: <GithubSVG/>,
                                id: t('onboarding_wizard.plugins.github'),
                                defaultMessage: 'Github',
                                checked: props.options.github,
                                tooltip: formatMessage({
                                    id: 'onboarding_wizard.plugins.github.tooltip',
                                    defaultMessage: 'Subscribe to repositories, stay up to date with reviews, assignments',
                                }),
                            },
                            {
                                onClick: () => props.setOption('gitlab'),
                                icon: <GitlabSVG/>,
                                id: t('onboarding_wizard.plugins.gitlab'),
                                defaultMessage: 'Gitlab',
                                checked: props.options.gitlab,
                                tooltip: formatMessage({
                                    id: 'onboarding_wizard.plugins.gitlab.tooltip',
                                    defaultMessage: 'Gitlab tooltip',
                                }),
                            },
                            {
                                onClick: () => props.setOption('jira'),
                                icon: <JiraSVG/>,
                                id: t('onboarding_wizard.plugins.jira'),
                                defaultMessage: 'Jira',
                                checked: props.options.jira,
                                tooltip: formatMessage({
                                    id: 'onboarding_wizard.plugins.jira.tooltip',
                                    defaultMessage: 'Jira tooltip',
                                }),
                            },
                            {
                                onClick: () => props.setOption('zoom'),
                                icon: <ZoomSVG/>,
                                id: t('onboarding_wizard.plugins.zoom'),
                                defaultMessage: 'Zoom',
                                checked: props.options.zoom,
                                tooltip: formatMessage({
                                    id: 'onboarding_wizard.plugins.zoom.tooltip',
                                    defaultMessage: 'Zoom tooltip',
                                }),
                            },
                            {
                                onClick: () => props.setOption('todo'),
                                icon: <TodoSVG/>,
                                id: t('onboarding_wizard.plugins.todo'),
                                defaultMessage: 'To do',
                                checked: props.options.todo,
                                tooltip: formatMessage({
                                    id: 'onboarding_wizard.plugins.todo.tooltip',
                                    defaultMessage: 'To do tooltip',
                                }),
                            },
                        ]}
                    />
                }
                <div className='Plugins__marketplace'>
                    <FormattedMarkdownMessage
                        id='onboarding_wizard.plugins.marketplace'
                        defaultMessage='More tools can be added once your workspace is set up. To see all available integrations, **[visit the marketplace](https://mattermost.com/marketplace/)**'
                    />
                </div>
                <div>
                    <button
                        className='primary-button'
                        onClick={props.next}
                    >
                        <FormattedMessage
                            id={'onboarding_wizard.next'}
                            defaultMessage='Continue'
                        />
                    </button>
                    <button
                        className='tertiary-button'
                        onClick={props.skip}
                    >
                        <FormattedMessage
                            id={'onboarding_wizard.skip'}
                            defaultMessage='Skip for now'
                        />
                    </button>
                </div>
                <PageLine/>
            </div>
        </CSSTransition>
    );
};

export default Plugins;
