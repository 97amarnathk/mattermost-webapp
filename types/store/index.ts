// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {Channel} from 'mattermost-redux/types/channels';
import {Post} from 'mattermost-redux/types/posts';
import {GlobalState as BaseGlobalState} from 'mattermost-redux/types/store';
import {RelationOneToOne} from 'mattermost-redux/types/utilities';

import {PostDraft} from 'types/drafts';

import {PluginsState} from './plugins';
import {ViewsState} from './views';

export type DraggingState = {
    state?: string;
    type?: string;
    id?: string;
}

export type GlobalState = BaseGlobalState & {
    drafts: {
        commentDrafts: RelationOneToOne<Post, PostDraft>;
        postDrafts: RelationOneToOne<Channel, PostDraft>;
    };
    plugins: PluginsState;
    storage: {
        storage: Record<string, any>;
        initialized: boolean;
    };

    views: ViewsState;
};
