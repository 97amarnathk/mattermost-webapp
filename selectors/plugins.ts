// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {createSelector} from 'reselect';

import {GlobalState} from 'types/store';
import {FileDropdownPluginComponent, PluginComponent} from '../types/store/plugins';

export const getFilesDropdownPluginMenuItems = createSelector(
    'getFilesDropdownPluginMenuItems',
    (state: GlobalState) => state.plugins.components.FilesDropdown,
    (components) => {
        return (components || []) as unknown as FileDropdownPluginComponent[];
    },
);

export const getChannelHeaderPluginComponents = createSelector(
    'getChannelHeaderPluginComponents',
    (state: GlobalState) => state.plugins.components.ChannelHeaderButton,
    (components) => {
        return (components || []) as unknown as PluginComponent[];
    },
);
