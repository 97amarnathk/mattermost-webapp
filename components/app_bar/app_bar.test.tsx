// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import {Channel} from 'mattermost-redux/types/channels';
import {Theme} from 'mattermost-redux/types/themes';

import {PluginComponent} from 'types/store/plugins';

import AppBar, {Props} from './app_bar';

describe('components/app_bar/app_bar', () => {
    const channelHeaderIcon = <img id='the_icon'/>;

    const channelHeaderComponents: PluginComponent[] = [
        {
            id: 'the_component_id',
            pluginId: 'playbooks',
            icon: channelHeaderIcon,
            action: jest.fn(),
        },
    ];

    const baseProps: Props = {
        activePluginId: 'playbooks',
        appBarBindings: [],
        channelHeaderComponents,
        channel: {} as Channel,
        theme: {} as Theme,
        show: true,
    };

    test('should match snapshot on mount', async () => {
        const props = {
            ...baseProps,
        };

        const wrapper = shallow<AppBar>(
            <AppBar {...props}/>,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should show provided icon for each channel header component', async () => {
        const props = {
            ...baseProps,
        };

        const wrapper = shallow<AppBar>(
            <AppBar {...props}/>,
        );

        expect(wrapper.find(channelHeaderIcon).exists()).toBe(false);
    });
});
