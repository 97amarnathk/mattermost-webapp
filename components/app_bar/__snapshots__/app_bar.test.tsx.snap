// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/app_bar/app_bar should match snapshot on mount 1`] = `<AppBar />`;
